src/sentry_scope.c

```diff
@@ -110,6 +112,107 @@ sentry__scope_flush(const sentry_scope_t *scope)
     }
 }
 
+static void
+sentry__foreach_stacktrace(
+    sentry_value_t event, void (*func)(sentry_value_t stacktrace))
+{
+    // We have stacktraces at the following locations:
+    // * `exception[.values].X.stacktrace`:
+    //   https://develop.sentry.dev/sdk/event-payloads/exception/
+    // * `threads[.values].X.stacktrace`:
+    //   https://develop.sentry.dev/sdk/event-payloads/threads/
+
+    sentry_value_t exception = sentry_value_get_by_key(event, "exception");
+    if (sentry_value_get_type(exception) == SENTRY_VALUE_TYPE_OBJECT) {
+        exception = sentry_value_get_by_key(exception, "values");
+    }
+    if (sentry_value_get_type(exception) == SENTRY_VALUE_TYPE_LIST) {
+        size_t len = sentry_value_get_length(exception);
+        for (size_t i = 0; i < len; i++) {
+            sentry_value_t stacktrace = sentry_value_get_by_key(
+                sentry_value_get_by_index(exception, i), "stacktrace");
+            if (!sentry_value_is_null(stacktrace)) {
+                func(stacktrace);
+            }
+        }
+    }
+
+    sentry_value_t threads = sentry_value_get_by_key(event, "threads");
+    if (sentry_value_get_type(threads) == SENTRY_VALUE_TYPE_OBJECT) {
+        threads = sentry_value_get_by_key(threads, "values");
+    }
+    if (sentry_value_get_type(threads) == SENTRY_VALUE_TYPE_LIST) {
+        size_t len = sentry_value_get_length(threads);
+        for (size_t i = 0; i < len; i++) {
+            sentry_value_t stacktrace = sentry_value_get_by_key(
+                sentry_value_get_by_index(threads, i), "stacktrace");
+            if (!sentry_value_is_null(stacktrace)) {
+                func(stacktrace);
+            }
+        }
+    }
+}
+
+static void
+sentry__symbolize_frame(const sentry_frame_info_t *info, void *data)
+{
+    // See https://develop.sentry.dev/sdk/event-payloads/stacktrace/
+    sentry_value_t frame = *(sentry_value_t *)data;
+
+    if (info->symbol
+        && sentry_value_is_null(sentry_value_get_by_key(frame, "function"))) {
+        sentry_value_set_by_key(
+            frame, "function", sentry_value_new_string(info->symbol));
+    }
+
+    if (info->object_name
+        && sentry_value_is_null(sentry_value_get_by_key(frame, "package"))) {
+        sentry_value_set_by_key(
+            frame, "package", sentry_value_new_string(info->object_name));
```

Comment:
_nit_: If you want to cut down on size, you could also remove this. It definitely never hurts to send the `package`, but in Sentry we will restore or even override this attribute from the `debug_images` list.

Reply:
So on my windows when I tried this earlier, the `package` had all the absolute filesystem paths, so this can get quite large. I will leave this in for now, as especially on linux, we don’t have automatic reloading of the `debug_images`.